packages/data-stores/src/wpcom-request-controls/index.ts

```diff
@@ -26,7 +26,19 @@ function triggerWpcomRequest( params: WpcomProxyRequestOptions ): Promise< unkno
 	} );
 }
 
-export const controls = {
-	WPCOM_REQUEST: ( { request }: ReturnType< typeof wpcomRequest > ) =>
-		triggerWpcomRequest( request ),
-} as const;
+export function createControls( clientCreds?: WpcomClientCredentials ) {
```

Comment:
As far as I can tell, the client credentials are specific to a few endpoints and not so much a general concern. It seems like something the stores should handle.

Reply:
Hmm you're right. One way I think I can do this is to store the credentials in the store and then `SELECT` them before triggering a `WPCOM_REQUEST`. I don't really like the idea of credentials in the store though, they never change and it's just duplication. But maybe I'm being too purest.

Perhaps I could have a `createActions()` factory that acceptst the credentials as params. I'll give that a go, but I'm wondering if it'll making the typings worse.

Reply:
Getting the types to work with a `createActions()` factory wasn't too hard after all.